stockflux-securities-master/src/components/TextField.js

```diff
@@ -2,22 +2,22 @@ import React from "react";
 import "./TextField.css";
 import PropTypes from "prop-types";
 
-const TextField = ({ readOnly, className, ...props }) => {
+const TextField = ({ readOnly, id, value, onChange }) => {
   return (
     <input
-      {...props}
       readOnly={readOnly}
-      className={`input-form-input${readOnly ? " read-only" : ""}${
-        className ? ` ${className}` : ""
-      }`}
+      id={id}
+      value={value}
+      onChange={onChange}
+      className={`input-form-input ${readOnly && "read-only"}`}
     />
   );
 };
 
 TextField.propTypes = {
-  onChange: PropTypes.func,
   readOnly: PropTypes.bool,
-  className: PropTypes.string
+  value: PropTypes.string.isRequired,
```

Comment:
I agree that value should be a string, but should it be required? What happens in the case of an add form where there isn't a value? What about in the event that you re-use this component in another place in the future where there's no default value, like a search box or something? I think this should be optional.

Reply:
Okay, I've made this optional now